### PR TITLE
Add hidden columns to information_schema tables to make Hue happy

### PR DESCRIPTION
**Pull request by kokosing**
Add hidden columns to information_schema tables to make Hue happy

### CODE REVIEW

**Comment by kokosing** [comment]
Using Presto from Hue fails due the missing TABLE_COMMENT and MYSQL_COMMENT columns. It looks like Hue generic jdbc was only tested on MySQL.

**Comment by electrum** [comment]
I’d rather say “MySQL compatible” rather than mention a specific application

Have you tried filing a bug against Hue?

File: presto-main/src/main/java/io/prestosql/connector/informationSchema/InformationSchemaMetadata.java [hunk 1]
```diff
@@ -89,12 +89,14 @@
                     .column("data_type", createUnboundedVarcharType())
                     .column("comment", createUnboundedVarcharType())
                     .column("extra_info", createUnboundedVarcharType())
+                    .hiddenColumn("column_comment", createUnboundedVarcharType()) // this column is added for Hue
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/connector/informationSchema/InformationSchemaMetadata.java]: I'd change this to
```java
// MySQL compatible

File: presto-main/src/main/java/io/prestosql/connector/informationSchema/InformationSchemaMetadata.java [hunk 2]
```diff
@@ -89,12 +89,14 @@
...
                     .build())
             .table(tableMetadataBuilder(TABLE_TABLES)
                     .column("table_catalog", createUnboundedVarcharType())
                     .column("table_schema", createUnboundedVarcharType())
                     .column("table_name", createUnboundedVarcharType())
                     .column("table_type", createUnboundedVarcharType())
+                    .hiddenColumn("table_comment", createUnboundedVarcharType()) // this column is added for Hue
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/connector/informationSchema/InformationSchemaMetadata.java]: Same
```java
// MySQL compatible

File: presto-main/src/main/java/io/prestosql/connector/informationSchema/InformationSchemaPageSourceProvider.java [hunk 3]
```diff
@@ -160,7 +160,8 @@ private InternalTable buildColumns(Session session, Set<QualifiedTablePrefix> pr
                             "YES",
                             column.getType().getDisplayName(),
                             column.getComment(),
-                            column.getExtraInfo());
+                            column.getExtraInfo(),
+                            column.getComment() /* hidden column_comment column */);
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/connector/informationSchema/InformationSchemaPageSourceProvider.java]: I don't think this comment is needed (we don't have it for any others)

File: presto-main/src/main/java/io/prestosql/connector/informationSchema/InformationSchemaPageSourceProvider.java [hunk 4]
```diff
@@ -182,7 +183,8 @@ private InternalTable buildTables(Session session, Set<QualifiedTablePrefix> pre
                         prefix.getCatalogName(),
                         name.getSchemaName(),
                         name.getTableName(),
-                        type);
+                        type,
+                        null /* hidden table_comment column */);
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/connector/informationSchema/InformationSchemaPageSourceProvider.java]: I'd remove this and add a comment above
```java
// We provide a "table_comment" column for compatibility with MySQL, but do not
// populate it as fetching metadata per table is much more expensive than listing.

File: presto-tests/src/test/java/io/prestosql/tests/TestLocalQueries.java [hunk 5]
```diff
@@ -142,4 +142,14 @@ public void testIOExplain()
                 jsonCodec(IoPlan.class).fromJson((String) getOnlyElement(result.getOnlyColumnAsSet())),
                 new IoPlan(ImmutableSet.of(input), Optional.empty()));
     }
+
+    @Test
+    public void testHueQueries()
```
Comment (electrum) [inline presto-tests/src/test/java/io/prestosql/tests/TestLocalQueries.java]: Similarly, how about changing to `testMySqlInformationSchemaCompatibility` and removing the Hue links

**Comment by findepi** [comment]
> Similarly, how about changing to `testMySqlInformationSchemaCompatibility` and removing the Hue links

Making information_schema MySQL-compatible would involve adding more columns and more information_schema tables.

File: presto-tests/src/test/java/io/prestosql/tests/TestLocalQueries.java (hunk 5 above)
Comment (electrum) [reply presto-tests/src/test/java/io/prestosql/tests/TestLocalQueries.java]: Makes sense.

**Comment by kokosing** [comment]
I posted an issue for Hue, about this: https://github.com/cloudera/hue/issues/799

**Comment by kokosing** [comment]
And a fix: https://github.com/cloudera/hue/pull/800

File: presto-tests/src/test/java/io/prestosql/tests/TestLocalQueries.java (hunk 5 above)
Comment (kokosing) [reply presto-tests/src/test/java/io/prestosql/tests/TestLocalQueries.java]: So I am going to keep it as it is.

**Pull request merged**